eva-accession-import-automation/genome_downloader.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2020 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from get_assembly_report_url import *
+import os
+import pandas
+import sys
+import urllib.request
+import re
+import json
+from __init__ import *
+
+
+def get_header_line_index(assembly_report_path):
+    try:
+        with open(assembly_report_path) as assembly_report_file_handle:
+            return next(line_index for line_index, line in enumerate(assembly_report_file_handle)
+                        if line.lower().startswith("# sequence-name") and "sequence-role" in line.lower())
+    except StopIteration:
+        raise Exception("Could not locate header row in the assembly report!")
+
+
+def get_dataframe_for_assembly_report(assembly_report_path):
+    return pandas.read_csv(assembly_report_path, skiprows=get_header_line_index(assembly_report_path),
```

Comment:
Suggest using the chunksize parameter for read_csv to account for large assembly reports (unfortunately, there are some which have hundreds of thousands of contigs)